akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1050,38 +1056,38 @@ final case class MapAsyncUnordered[In, Out](parallelism: Int, f: In ⇒ Future[O
 
       override def preStart(): Unit = buffer = BufferImpl(parallelism, materializer)
 
-      private val futureCB =
-        getAsyncCallback((result: Try[Out]) ⇒ {
-          inFlight -= 1
-          result match {
-            case Success(elem) if elem != null ⇒
-              if (isAvailable(out)) {
-                if (!hasBeenPulled(in)) tryPull(in)
-                push(out, elem)
-              } else buffer.enqueue(elem)
-            case other ⇒
-              val ex = other match {
-                case Failure(t)              ⇒ t
-                case Success(s) if s == null ⇒ ReactiveStreamsCompliance.elementMustNotBeNullException
-              }
-              if (decider(ex) == Supervision.Stop) failStage(ex)
-              else if (isClosed(in) && todo == 0) completeStage()
-              else if (!hasBeenPulled(in)) tryPull(in)
-          }
-        }).invoke _
+      def futureCompleted(result: Try[Out]): Unit = {
+        inFlight -= 1
+        result match {
+          case Success(elem) if elem != null ⇒
+            if (isAvailable(out)) {
+              if (!hasBeenPulled(in)) tryPull(in)
+              push(out, elem)
+            } else buffer.enqueue(elem)
+          case other ⇒
+            val ex = other match {
+              case Failure(t)              ⇒ t
+              case Success(s) if s == null ⇒ ReactiveStreamsCompliance.elementMustNotBeNullException
```

Comment:
this pattern happens a few times already, useful enough to have method on ReactiveStreamsCompliance for it?